build.sbt

```diff
@@ -804,60 +270,71 @@ lazy val testkit = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .settings(moduleName := "cats-testkit")
-  .settings(catsSettings)
+  .settings(commonSettings)
+  .settings(tlFatalWarnings := false)
   .settings(disciplineDependencies)
   .jsSettings(commonJsSettings)
-  .jvmSettings(commonJvmSettings ++ mimaSettings("cats-testkit", includeCats1 = false))
-  .settings(scalacOptions := scalacOptions.value.filter(_ != "-Xfatal-warnings"))
+  .jvmSettings(commonJvmSettings)
   .nativeSettings(commonNativeSettings)
 
 lazy val alleycatsCore = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .crossType(CrossType.Pure)
   .in(file("alleycats-core"))
   .dependsOn(core)
   .settings(moduleName := "alleycats-core", name := "Alleycats core")
-  .settings(catsSettings)
-  .settings(publishSettings)
-  .settings(includeGeneratedSrc)
+  .settings(commonSettings)
   .jsSettings(commonJsSettings)
-  .jvmSettings(commonJvmSettings ++ mimaSettings("alleycats-core", includeCats1 = false))
+  .jvmSettings(commonJvmSettings)
   .nativeSettings(commonNativeSettings)
 
 lazy val alleycatsLaws = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .crossType(CrossType.Pure)
   .in(file("alleycats-laws"))
   .dependsOn(alleycatsCore, laws)
   .settings(moduleName := "alleycats-laws", name := "Alleycats laws")
-  .settings(catsSettings)
-  .settings(publishSettings)
+  .settings(commonSettings)
   .settings(disciplineDependencies)
   .settings(testingDependencies)
   .jsSettings(commonJsSettings)
-  .jvmSettings(commonJvmSettings ++ mimaSettings("alleycats-laws", includeCats1 = false))
+  .jvmSettings(commonJvmSettings)
   .nativeSettings(commonNativeSettings)
 
 lazy val alleycatsTests = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .in(file("alleycats-tests"))
   .dependsOn(alleycatsLaws, tests % "test-internal -> test")
+  .enablePlugins(NoPublishPlugin)
   .settings(moduleName := "alleycats-tests")
-  .settings(catsSettings)
-  .settings(noPublishSettings)
+  .settings(commonSettings)
+  .settings(Test / scalacOptions := (Test / scalacOptions).value.filter(_ != "-Xfatal-warnings"))
   .jsSettings(commonJsSettings)
   .jvmSettings(commonJvmSettings)
-  .settings(Test / scalacOptions := (Test / scalacOptions).value.filter(_ != "-Xfatal-warnings"))
   .nativeSettings(commonNativeSettings)
 
+lazy val unidocs = project
+  .enablePlugins(TypelevelUnidocPlugin)
+  .settings(
+    name := "cats-docs",
+    ScalaUnidoc / unidoc / unidocProjectFilter := inProjects(kernel.jvm,
+                                                             core.jvm,
+                                                             free.jvm,
+                                                             algebra.jvm,
+                                                             alleycatsCore.jvm
+    ),
+    scalacOptions ~= { _.filterNot(_.startsWith("-W")) }, // weird nsc bug
+    ScalaUnidoc / unidoc / scalacOptions ++= Seq("-groups", "-diagrams")
+  )
+  .settings(commonSettings)
```

Comment:
This introduces a new API-docs only `cats-docs` module that lets us publish unidocs to Sonatype/Maven. Then they can be browsed at javadoc.io.

Example:
https://www.javadoc.io/doc/org.http4s/http4s-docs_2.13/0.23.11/org/http4s/index.html